Update quickstart.md (#577)
Redocly/openapi-cli: docs/quickstart.md

# Redocly OpenAPI CLI quickstart guide

## Before you begin

* Get yourself a [GitHub](https://github.com/) account.
* Install [node.js](https://nodejs.org/en/) on your local computer.

## Step 1 - Clone the openapi-starter project

[openapi-starter](https://github.com/Redocly/openapi-starter) creates a local folder structure that you can use to manage your API definition either as a single file, or in multi-file format. It also places a basic API definition file called `openapi.yaml` in the `openapi` root folder for those of you who don't have a definition file but need one to start exploring OpenAPI CLI. You're welcome!

If you have your own definition file, go right ahead and use it.

Go [here](openapi-starter.md) to learn how to clone the `openapi-starter` project and learn more about how it works.

## Step 2 - Install OpenAPI CLI from openapi-starter

> Run `npm install` in the root of the `openapi-starter` repo.

Reviewer: ```suggestion
Run `npm install` in the root of your local `openapi-starter` project.
```

Reviewer: I'm suggesting this to make it clear that this won't be done on the actual repo but on the user's own copy.

## Step 3 - Prepare your API definition file

In this guide, we'll use `openapi.yaml` in all examples. We recommend that you use it until you're confident with OpenAPI CLI.

## Step 4 - Try some basic commands

> [Commands](./docs/commands/index.md) are used to run tasks (like splitting up large definition files and putting them back together) and they can also return information (like getting stats about your definition). To get you started, we'll focus on the top four: `bundle`, `lint`, `split` and `preview`.

Reviewer: ```suggestion

[Commands](./docs/commands/index.md) are used to run tasks (like splitting up large definition files and putting them back together) and they can also return information (like getting stats about your definition). To get you started, we'll focus on the top four: `bundle`, `lint`, `split` and `preview`.
```

:::success Tip

> If you're new to command lines, you don't have to include the `$` in your commands. Just open the node.js command window, make sure you `cd` to the path where your API definition file is stored, type the command then press **Enter**.

Reviewer: ```suggestion
If you're new to command-line interfaces, note that you don't have to include the `$` in your commands. This symbol only indicates that the command after it should be entered into the command prompt. To run a command, open your command prompt, use `cd` to access the path where your API definition file is stored, type the command, and press **Enter**.
```

:::

> ### `bundle` - Pull constituent parts of your definition back into a single file

Reviewer: issue: This heading is quite long - I think it's the longest of all on this page (the previous one had the same problem). This might cause issues with the table of contents on the right side of the page. It won't necessarily break anything, but it will look awkward. Do you have any ideas for how we could shorten it? If not, it's fine, not a blocker - we can leave it and maybe get back to it later.

> `bundle` merges standalone files back into a single definition file.

Reviewer: Suggesting this to have a somewhat smoother intro to the sentence. Removing "back" because it implies that you have previously split the definition, which doesn't always have to be the case. You can have the multi-file definition from the get-go and won't have to split anything; you will just want to bundle it.

> `bundle` merges standalone files back into a single definition file.

Reviewer: Actually, could we add another sentence to explain this context, and to make it clear that a multi-file structure must exist for the `bundle` command to fulfill its purpose? It can be a pre-existing multi-file structure, or the user can create one with the `split` command. However, in the latter case, we would have to reorder the commands and introduce split before bundle... :thinking:

In your terminal, type the following command. We'll output the constituent files into a new folder called `bundled` to keep things neat and tidy, but you could also specify an existing folder:

```bash
$ openapi split bundled.yaml --outDir bundled
```

You should get this response:

```bash
Document: bundled.yaml is successfully split
  and all related files are saved to the directory: bundled

bundled.yaml: split processed in 76ms
```

So, what just happened? Take a look in the newly created `bundled` directory. That's where the magic is! See how the `split` command automatically broke the single API definition into its constituent parts and very kindly organized them in the new directory?

```bash
├── code samples
│   └── C#
│       └── postundefined
│   └── PHP
│       └── postundefined
├── components
│   └── headers
│       └── ExpiresAfter.yaml
│   └── responses
│       └── Problem.yaml
│   └── schemas
│       └── Email.yaml
│       └── Problem.yaml
│       └── Schema.yaml
│       └── User.yaml
├── paths
│       └── echo.yaml
│       └── pathItem.yaml
│       └── pathItemWithExamples.yaml
│       └── users@{username}.yaml
└── openapi.yaml
```

:::info Note

> `openapi.yaml` in the `bundled` folder is the default name that Open API CLI gives to the 'master' YAML file you just split up. It contains `$ref`s to its constituent parts. You do the work in the consituent parts.

Reviewer: ```suggestion
`openapi.yaml` in the `bundled` folder is the default name that OpenAPI CLI gives to the 'root' YAML file you just split up. It contains `$ref`s to its constituent parts. You do the work in the consituent parts.
```

Reviewer: Suggesting to fix some typos and phrasing

# Redocly OpenAPI CLI quickstart guide

## Before you begin

* Get yourself a [GitHub](https://github.com/) account.
* Install [node.js](https://nodejs.org/en/) on your local computer.

## Step 1 - Clone the openapi-starter project

[openapi-starter](https://github.com/Redocly/openapi-starter) creates a local folder structure that you can use to manage your API definition either as a single file, or in multi-file format. It also places a basic API definition file called `openapi.yaml` in the `openapi` root folder for those of you who don't have a definition file but need one to start exploring OpenAPI CLI. You're welcome!

If you have your own definition file, go right ahead and use it.

Go [here](openapi-starter.md) to learn how to clone the `openapi-starter` project and learn more about how it works.

> 

Reviewer 2: **suggestion**: Maybe use a more descriptive word such as Go to the `openapi-starter README` to learn..

## Step 2 - Install OpenAPI CLI from openapi-starter

Run `npm install` in the root of the `openapi-starter` repo.

## Step 3 - Prepare your API definition file

> In this guide, we'll use `openapi.yaml` in all examples. We recommend that you use it until you're confident with OpenAPI CLI.

Reviewer: ```suggestion
In this guide, we'll use `openapi.yaml` from the `openapi-starter` project in all examples. We recommend that you use it until you're confident with OpenAPI CLI.
```

## Before you begin

* Get yourself a [GitHub](https://github.com/) account.
* Install [node.js](https://nodejs.org/en/) on your local computer.

## Step 1 - Clone the openapi-starter project

[openapi-starter](https://github.com/Redocly/openapi-starter) creates a local folder structure that you can use to manage your API definition either as a single file, or in multi-file format. It also places a basic API definition file called `openapi.yaml` in the `openapi` root folder for those of you who don't have a definition file but need one to start exploring OpenAPI CLI. You're welcome!

If you have your own definition file, go right ahead and use it.

If you want to learn how to clone the `openapi-starter` project and see how it works, we have [an entire page dedicated to it](openapi-starter.md)!

## Step 2 - Install OpenAPI CLI from openapi-starter

Run `npm install` in the root of your local `openapi-starter` project.

## Step 3 - Prepare your API definition file

In this guide, we'll use `openapi.yaml` from the `openapi-starter` project in all examples. We recommend that you use it until you're confident with OpenAPI CLI.

## Step 4 - Try some basic commands

> [OpenAPI CLI commands](./docs/commands/index.md) are used to run tasks (like splitting up large definition files and putting them back together) and they can also return information (like getting stats about your definition). To get you started, we'll focus on the top four: `bundle`, `lint`, `split` and `preview`.

Reviewer: ```suggestion
[OpenAPI CLI commands](./commands/index.md) are used to run tasks (like splitting up large definition files and putting them back together) and they can also return information (like getting stats about your definition). To get you started, we'll focus on the top four: `bundle`, `lint`, `split` and `preview`.
```

# Redocly OpenAPI CLI quickstart guide

## Before you begin

* Get yourself a [GitHub](https://github.com/) account.
* Install [node.js](https://nodejs.org/en/) on your local computer.

## Step 1 - Install OpenAPI CLI

[Follow these steps to install OpenAPI CLI](./docs/installation.md).

## Step 2 - Clone the openapi-starter project

[openapi-starter](https://github.com/Redocly/openapi-starter) creates a local folder structure that you can use to manage your API definition either as a single file, or in multi-file format. It also places a basic API definition file in the `openapi` root folder for those of you who don't have a definition file but need one to start exploring OpenAPI CLI. You're welcome!

If you have your own definition file, go right ahead and use it.

Go [here](openapi-starter.md) to learn how to clone the `openapi-starter` project and learn more about how it works.

> ## Step 3 - Prepare your API definition file

Reviewer: suggestion: I think it would make more sense to continue using the example API definition from the openapi-starter repo here, instead of switching to a completely different, third-party API definition. I know you say users can choose whichever they want, but we've just spent an entire step explaining how to set up openapi-starter, only to seemingly abandon it in the next step.

Author: @skadinna - you would like me to abandon using openweathermap.yaml altogether and just use the example API definition throughout for all examples?

Reviewer: I _think_ it would be easier for both the readers and the docs writers, and here's why:

- if we use it, we're telling the readers a cohesive story where all our stuff is more or less integrated. That way, they always know where to find the complete API definition - in the openapi-starter repo. Also, if they find something wrong in the API definition or want to propose some improvements to it, they could do it by opening an issue in the openapi-starter repo. No such option when we use someone else's API definition. If they ask us e.g. why some schema looks the way it looks in a particular example, we can't really answer that for someone else's API...
- continuing from the previous point, if we use an API definition that we fully control, docs writers always know who to ask for help, and they will also know when the API definition is updated. If we use someone else's API definition, the writers would either have to periodically check whether the original authors have updated something in the API definition, or just not care about the definition being up-to-date. 

Of course, this is just my opinion...I hope it makes at least some sense :sweat_smile: Might be good to wait for feedback from @adamaltman and @swapnilogale too, before you proceed with any changes.

Author: No, Ivana that makes _perfect_ sense. :) I only introduced the openweathermap.yaml as I know you guys were trying to move away from petstore! If Adam and Swapnil agree, then I will re-do the examples using openapi.yaml.

Reviewer 3: I agree to use the openapi-starter repo definition. 👍

Reviewer 2: Agree with what @skadinna said. It's easier to control expectations with our own API definition, rather than people grumble about something we have no control over. Also, it allows us to identify any shortcomings with our own API.

          example: 6
    Sys:
      title: Sys
      type: object
      properties:
        type:
          type: integer
          description: Internal parameter
          format: int32
          example: 1
        id:
          type: integer
          description: Internal parameter
          format: int32
          example: 8166
        message:
          type: number
          description: Internal parameter
          example: 0.0166
        country:
          type: string
          description: Country code (GB, JP etc.)
          example: AU
        sunrise:
          type: integer
          description: Sunrise time, unix, UTC
          format: int32
          example: 1435610796
        sunset:
          type: integer
          description: Sunset time, unix, UTC
          format: int32
          example: 1435650870

  securitySchemes:
    app_id:
      type: apiKey
      description: "API key to authorize requests. (If you don't have an API key, get one at https://openweathermap.org/. See https://idratherbewriting.com/learnapidoc/docapis_get_auth_keys.html for details.)"
      name: appid
      in: query
```

## Step 4 - Try some basic commands

> [Commands](./docs/commands/index.md) are used to run tasks (like splitting up large definition files and putting them back together) and they can also return information (like getting stats about your definition). To get your started, we'll focus on the top four: `lint`, `split`, `bundle` and `preview`.

Reviewer: ```suggestion
[Commands](./docs/commands/index.md) are used to run tasks (like splitting up large definition files and putting them back together) and they can also return information (like getting stats about your definition). To get you started, we'll focus on the top four: `lint`, `split`, `bundle` and `preview`.
```
